lib/router.js

```diff
@@ -36,10 +36,15 @@ Cherrytree.prototype.initialize = function (options) {
   this.middleware = []
   this.options = _.extend({
     interceptLinks: true,
-    logError: true
+    logError: true,
+    Promise: window.Promise
   }, options)
   this.log = createLogger(this.options.log)
   this.logError = createLogger(this.options.logError, true)
+
+  if (typeof this.options.Promise !== 'function') {
```

Comment:
Is `invariant` new ES6 (7?) sugar for `if (!....) The new Error(...)`? I couldn't find any documentation on it.